ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -1506,207 +1506,221 @@ static IHqlExpression * createArith(node_operator op, ITypeInfo * type, IHqlExpr
+
+                //average should be done as a real operation I think, possibly decimal, if argument is decimal
+                OwnedHqlExpr n1 = createArith(no_mul, exprType, sumx, sumy);
+                OwnedHqlExpr n2 = createArith(no_div, exprType, n1, count);
+                OwnedHqlExpr n3 = createArith(no_sub, exprType, sumxy, n2);
+                OwnedHqlExpr n4 = createArith(no_div, exprType, n3, count);
+                return normalizeAggregateExpr(selector, n4, false);
+            }
+        case no_corrgroup:
+            //Map this to (covar(x,y)/(var(x).var(y)))
+            //== (sum(x.y)*count() - sum(x).sum(y))/sqrt((sum(x.x)*count()-sum(x)^2) * (sum(y.y)*count()-sum(y)^2))
+            {
+                IHqlExpression * argX = expr->queryChild(0);
+                IHqlExpression * argY = expr->queryChild(1);
+                IHqlExpression * cond = expr->queryChild(2);
+
+                ITypeInfo * exprType = expr->queryType();
+                OwnedHqlExpr xx = createArith(no_mul, exprType, argX, argX);
+                OwnedHqlExpr sumxx = createValue(no_sumgroup, LINK(exprType), LINK(xx), LINK(cond));
+                OwnedHqlExpr xy = createArith(no_mul, exprType, argX, argY);
+                OwnedHqlExpr sumxy = createValue(no_sumgroup, LINK(exprType), LINK(xy), LINK(cond));
+                OwnedHqlExpr yy = createArith(no_mul, exprType, argY, argY);
+                OwnedHqlExpr sumyy = createValue(no_sumgroup, LINK(exprType), LINK(yy), LINK(cond));
+                OwnedHqlExpr sumx = createValue(no_sumgroup, LINK(exprType), LINK(argX), LINK(cond));
+                OwnedHqlExpr sumy = createValue(no_sumgroup, LINK(exprType), LINK(argY), LINK(cond));
+                OwnedHqlExpr count = createValue(no_countgroup, LINK(defaultIntegralType), LINK(cond));
+
+                OwnedHqlExpr n1 = createArith(no_mul, exprType, sumxy, count);
+                OwnedHqlExpr n2 = createArith(no_mul, exprType, sumx, sumy);
+                OwnedHqlExpr n3 = createArith(no_sub, exprType, n1, n2);
+
+                OwnedHqlExpr n4 = createArith(no_mul, exprType, sumxx, count);
+                OwnedHqlExpr n5 = createArith(no_mul, exprType, sumx, sumx);
+                OwnedHqlExpr n6 = createArith(no_sub, exprType, n4, n5);
+
+                OwnedHqlExpr n7 = createArith(no_mul, exprType, sumyy, count);
+                OwnedHqlExpr n8 = createArith(no_mul, exprType, sumy, sumy);
+                OwnedHqlExpr n9 = createArith(no_sub, exprType, n7, n8);
+
+                OwnedHqlExpr n10 = createArith(no_mul, exprType, n6, n9);
+                OwnedHqlExpr n11 = createValue(no_sqrt, LINK(exprType), LINK(n10));
+                OwnedHqlExpr n12 = createArith(no_div, exprType, n3, n11);
+                return normalizeAggregateExpr(selector, n12, false);
+            }
+            throwUnexpected();
+
+        case no_variance:
+        case no_covariance:
+        case no_correlation:
+        case no_field:
```

Comment:
Do we want a test for this or are there any that now hit this? I presume all the reg suits were run and passed ok?